cogs/moderation.py

```diff
@@ -1,29 +1,57 @@
 import asyncio
+from collections import Counter
+import typing as t
+from .utils.formats import plural
 import textwrap
 
 import discord
 from discord import Color, Embed
 from discord.ext.commands import (
+    BadArgument,
     Bot,
     Cog,
     Context,
+    Converter,
     bot_has_permissions,
     command,
+    guild_only,
+    Greedy,
     has_permissions,
+    MemberConverter,
 )
 
 
-class Moderation(Cog):
-    """This cog provides moderation commands."""
+def can_execute_action(ctx: Context, user: discord.User, target: discord.User) -> bool:
+    return user.id == ctx.bot.owner_id or user == ctx.guild.owner or user.top_role > target.top_role
+
+
+class MemberNotFound(Exception):
+    pass
+
+
+async def resolve_member(self, guild: discord.Guild, member_id: int) -> discord.User:
+    user = guild.get_member(member_id)
+    if user is None:
+        if guild.chunked:
+            raise MemberNotFound()
+        try:
+            user = await guild.fetch_member(member_id)
+        except discord.NotFound:
+            raise MemberNotFound()
+    return user
 
-    def __init__(self, bot: Bot) -> None:
-        self.bot = bot
+
+# TODO : Clean everything up
+class Moderation(Cog):
+    def __init__(self, client: discord.Bot) -> None:
+        self.client = client
 
     @command()
     @bot_has_permissions(kick_members=True)
     @has_permissions(kick_members=True)
-    async def kick(self, ctx: Context, member: discord.Member, *, reason: str = "No reason.") -> None:
+    async def kick(self, ctx: Context, member: discord.Member, *, reason: str = "No specific reason") -> None:
```

Comment:
Why limit ourselves with only using `discord.Member` when you created a custom converter for `MemberID`. Use that but note that you'll need to check if the user which comes from it is Member or not if he isn't a member, this should give an error, as we can't kick, non-connected users

Reply:
idk this

Reply:
I'm not sure what's hard on that, just change type from `discord.Member` to `UserID` and check if the resulting member in `member` is a part of that guild.